### ISSUE TITLE
docker run - npm error

### ISSUE THREAD

**Issue by shachar058** [openemr/openemr#3297]
### Describe the bug

When running a docker container from the openemr/openemr:flex-3.10 image, I first get the following warnings:
```
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/formats/bps.js'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/gulp-cli-1f5e74ff/lib/shared/ansi.js'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/jquery-datetimepicker-82631704/screen/6.png'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/formats/bytes.js'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/gulp-cli-1f5e74ff/lib/shared/cli-options.js'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/vinyl-fs-aa250fce/lib/src/read-contents/index.js'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/formats/currency.js'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/gulp-cli-1f5e74ff/lib/shared/completion.js'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/dropzone-07fd6a3a/dist/min'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/vinyl-fs-aa250fce/lib/src/read-contents/read-buffer.js'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/formats/exponential.js'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/vinyl-fs-aa250fce/lib/src/read-contents/read-dir.js'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/formats/ordinal.js'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/dropzone-07fd6a3a/dist/min'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/vinyl-fs-aa250fce/lib/src/read-contents/read-stream.js'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/formats/percentage.js'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/vinyl-fs-aa250fce/lib/src/read-contents/read-symbolic-link.js'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/formats/time.js'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/dropzone-07fd6a3a/dist/min'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/vinyl-fs-aa250fce/lib/src/resolve-symlinks.js'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/vinyl-fs-aa250fce/lib/src/sourcemap.js'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/dropzone-07fd6a3a/dist/min'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/chart.js-dd8608f8/dist/Chart.bundle.min.js'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/vinyl-fs-aa250fce/lib/src/wrap-vinyl.js'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/vinyl-fs-aa250fce/lib/symlink'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/bootstrap-v4-rtl-f7c2b7f4/dist/css/bootstrap.min.css'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/@fortawesome/fontawesome-free-47004691/css/v4-shims.min.css'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/vinyl-fs-aa250fce/lib/symlink'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/select2-5f634633/src/js/select2/dropdown/infiniteScroll.js'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/bootswatch-f82f2147/dist/cosmo/bootstrap.min.css'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/select2-5f634633/src/js/select2/compat/initSelection.js'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/vinyl-fs-aa250fce/lib/symlink'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/select2-5f634633/src/js/select2/compat/inputData.js'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/ckeditor4-70ac2613/composer.json'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/jspdf-ab5b0c9e/dist/jspdf.node.debug.js'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/ckeditor4-70ac2613/config.js'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/vinyl-fs-aa250fce/lib/symlink'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/ckeditor4-70ac2613/contents.css'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/konva-bd4bd85d/konva.min.js'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/ckeditor4-70ac2613/LICENSE.md'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/bootstrap-22516293/dist/js/bootstrap.bundle.js'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/select2-5f634633/src/js/select2/i18n/ja.js'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/select2-5f634633/src/js/select2/i18n/ka.js'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/summernote-4d17c3e7/dist/summernote.min.js.map'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/select2-5f634633/src/js/select2/keys.js'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/select2-5f634633/src/js/select2/i18n/km.js'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/select2-5f634633/src/js/select2/i18n/ko.js'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/select2-5f634633/src/js/select2/i18n/lt.js'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/select2-5f634633/src/js/select2/i18n/lv.js'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/select2-5f634633/src/js/select2/compat/matcher.js'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/select2-5f634633/src/js/select2/data/maximumInputLength.js'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/select2-5f634633/src/js/select2/data/maximumSelectionLength.js'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/select2-5f634633/src/js/select2/data/minimumInputLength.js'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/select2-5f634633/src/js/select2/dropdown/minimumResultsForSearch.js'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/select2-5f634633/src/js/select2/i18n/mk.js'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/select2-5f634633/src/js/select2/i18n/ms.js'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/select2-5f634633/src/js/select2/selection/multiple.js'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/select2-5f634633/src/js/select2/i18n/nb.js'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/dwv-b1198e1f/dist/dwv.min.js'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/select2-5f634633/src/js/select2/i18n/ne.js'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/select2-5f634633/src/js/select2/i18n/nl.js'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/select2-5f634633/src/js/select2/options.js'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/select2-5f634633/src/js/select2/i18n/pl.js'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/select2-5f634633/src/js/select2/selection/placeholder.js'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/select2-5f634633/src/js/select2/i18n/ps.js'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/select2-5f634633/src/js/select2/i18n/pt-BR.js'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/select2-5f634633/src/js/select2/i18n/pt.js'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/select2-5f634633/src/js/select2/compat/query.js'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/select2-5f634633/src/js/select2/results.js'
npm WARN tar ENOENT: no such file or directory, lstat '/var/www/localhost/htdocs/openemr/node_modules/.staging/numeral-7f77428c/src/locales'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/angular-5ee17fc2/angular.min.js'
npm WARN tar ENOENT: no such file or directory, open '/var/www/localhost/htdocs/openemr/node_modules/.staging/interactjs-351e6889/dist/api/fonts/OpenSans-Bold-webfont.svg'
npm WARN openemr-interface@0.2.0 No repository field.
npm WARN openemr-interface@0.2.0 license should be a valid SPDX license expression
```

And then I get the following error:
```
npm ERR! code 128
npm ERR! Command failed: git clone --mirror -q git://github.com/EastDesire/jscolor.git /root/.npm/_cacache/tmp/git-clone-3af8a522/.git
npm ERR! fatal: could not create leading directories of '/root/.npm/_cacache/tmp/git-clone-3af8a522/.git'
npm ERR! 
```

### To Reproduce

1. docker pull openemr/openemr:flex-3.10
2. docker run --env-file env_file openemr/openemr:flex-3.10

My env_file contains the following variables:
```
MYSQL_HOST=localhost
MYSQL_ROOT_USER=root
MYSQL_ROOT_PASS=pass
MYSQL_USER=openemr
MYSQL_PASS=openemr
MYSQL_DATABASE=openemr1
OE_USER=admin
OE_PASS=pass
FLEX_REPOSITORY=https://github.com/openemr/openemr.git
FLEX_REPOSITORY_BRANCH=master
```

**Comment by bradymiller** [comment]
hi @shachar058 ,

Thanks for the report. There must be an odd npm permission thing going on. It has cropped up in most recent alpine 3.10 and alpine 3.11 based openemr dockers.

This error is not showing up in the 5.0.3 (dev docker) docker build (alpine 3.11) or the easy development env use of flex dockers (in these case, the code for the docker is not "contained"). But is showing up in:
1. the case you showed above where using most recent codebase from github in a "contained' flex docker
2. the insane development environment

What I have noted in the broken environments is that the node_modules directory is empty which leads me to believe that there is a permission issue where npm can not write to node_modules.

**Comment by shachar058** [comment]
@bradymiller thank you for the reply,
I might be missing something here - what do you mean by the code being "contained" or "not contained"?

**Comment by bradymiller** [comment]
hi @shachar058 ,
There are 2 different main dockers.

1. Dockers that already include the codebase (ie. on Dockerbuild the codebase is brought in and built). For example, here is the Dockerfile for the 5.0.3-dev, which we build every couple weeks:
https://github.com/openemr/openemr-devops/blob/master/docker/openemr/5.0.3/Dockerfile
(this one has no npm build issues)

2. Then the second type of docker is the `flex` dockers, which can be run in several different modes. They do not include any codebase, so on the first run the either download codebase or use local codebase and also do the build (composer and npm). The 3 main modes are:
a. easy dev docker environment (uses local code with a bunch of crazy shared volumes): https://github.com/openemr/openemr/blob/master/docker-compose.yml#L20-L50
b. insane dev docker environment (uses local code): https://github.com/openemr/openemr/blob/master/contrib/util/docker/docker-compose.yml#L106-L117
c. other where it is downloading code: https://github.com/openemr/openemr-devops/blob/master/docker/openemr/flex-3.11/README.md (see the example docker-compose.yml section there)

At this point in the `flex` series, mode `a` is working, however mode `b` and mode `c` are not working because of this npm issue (which I am guessing is related to a odd permission issue with npm).

**Comment by shachar058** [comment]
@bradymiller  thanks, that really cleared things up for me

**Comment by bradymiller** [comment]
Turns out this is a bug in the version being used in Alpine 3.10/3.11 . I fixed the flex docker, by using npm from the alpine edge version, which I confirmed to work. This is the fix used (note just temporary until alpine 3.11 updates version of npm): https://github.com/openemr/openemr-devops/commit/98c07d59fb3dee47f5b544187d3e38d3fe97a700
(the `flex` docker has been built and uploaded to docker hub)
